packages/amplify-graphql-api-construct/src/__tests__/__functional__/auth-modes.test.ts

```diff
@@ -161,4 +161,133 @@ describe('auth modes', () => {
       });
     });
   });
+
+  it('renders with identity pool auth', () => {
+    verifySynth((stack) => {
+      const identityPool = new cognito.CfnIdentityPool(stack, 'TestIdentityPool', { allowUnauthenticatedIdentities: true });
+      const appsync = new iam.ServicePrincipal('appsync.amazonaws.com');
+      const authenticatedUserRole = new iam.Role(stack, 'AuthRole', { assumedBy: appsync });
+      const unauthenticatedUserRole = new iam.Role(stack, 'UnauthRole', { assumedBy: appsync });
+
+      new AmplifyGraphqlApi(stack, 'TestApi', {
+        definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+          type Todo @model @auth(rules: [{ provider: iam, allow: public }, { provider: iam, allow: private }]) {
+            description: String!
+          }
+        `),
+        authorizationModes: {
+          identityPoolConfig: {
+            identityPoolId: identityPool.logicalId,
+            authenticatedUserRole,
+            unauthenticatedUserRole,
+          },
+        },
+      });
+    });
+  });
+
+  it('renders with IAM access enabled', () => {
+    verifySynth((stack) => {
+      new AmplifyGraphqlApi(stack, 'TestApi', {
+        definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+          type Todo @model @auth(rules: [{ provider: iam, allow: public }, { provider: iam, allow: private }]) {
+            description: String!
+          }
+        `),
+        authorizationModes: {
+          iamConfig: {
+            enableIamAuthorizationMode: true,
+          },
+        },
+      });
+    });
+  });
+
+  it('throws if deprecated and un-deprecated IAM and identityPool settings are used', () => {
+    expect(() => {
+      verifySynth((stack) => {
+        const identityPool = new cognito.CfnIdentityPool(stack, 'TestIdentityPool', { allowUnauthenticatedIdentities: true });
+        const appsync = new iam.ServicePrincipal('appsync.amazonaws.com');
+        const authenticatedUserRole = new iam.Role(stack, 'AuthRole', { assumedBy: appsync });
+        const unauthenticatedUserRole = new iam.Role(stack, 'UnauthRole', { assumedBy: appsync });
+
+        new AmplifyGraphqlApi(stack, 'TestApi', {
+          definition: AmplifyGraphqlDefinition.fromString(/* GraphQL */ `
+            type Todo @model @auth(rules: [{ provider: iam, allow: public }, { provider: iam, allow: private }]) {
+              description: String!
+            }
+          `),
+          authorizationModes: {
+            iamConfig: {
+              identityPoolId: identityPool.logicalId,
+              authenticatedUserRole,
+              unauthenticatedUserRole,
+              enableIamAuthorizationMode: true,
+            },
+          },
+        });
+      });
+    }).toThrow(
+      "Cannot use deprecated 'authorizationModes.iamConfig' options with 'authorizationModes.identityPoolConfig' " +
```

Comment:
The error message is not clear. Can we change the wordings? Something like `Invalid authorization modes configuration provided. Deprecated IAM configuration cannot be used with identity pool configuration.`.

Reply:
Updated.